### DISCUSSION TITLE
Is it a syntax coloring bug? In inactive code block.

### DISCUSSION THREAD

**Discussion by Demonese** [microsoft/vscode-cpptools#7627] [Q&A] [answered]
* no problem
![image](https://user-images.githubusercontent.com/57629614/120090305-f30adb80-c133-11eb-8566-8b499b9d4af2.png)
* syntax coloring is wrong
![image](https://user-images.githubusercontent.com/57629614/120090309-fef69d80-c133-11eb-9059-0890205214c2.png)
* no problem
![image](https://user-images.githubusercontent.com/57629614/120090323-17ff4e80-c134-11eb-8c23-fdc5d0c80f0e.png)
* no problem
![image](https://user-images.githubusercontent.com/57629614/120090319-0ddd5000-c134-11eb-9fa3-c3a53c5cfbc0.png)

**Comment by sean-mcmanus** [comment, marked as answer]
Are you able to provide some code sample that repros this? I'm not reproing it with
```
void DebugPrint(const char*s, int, int);

int main()
{
    #define PRIu32 "d"
    #if !_WIN64
    DebugPrint("d" PRIu32 "d", 0, 1);
    #else
    DebugPrint("Win32Stream::SetSize(%" PRIu32 ")\n", 0 , 1);
    #endif
    int i;
    return 0;
}
```
![image](https://user-images.githubusercontent.com/19859882/120386427-7451a200-c2dd-11eb-9e93-8c84d867f8e1.png)

The definition of PRIu32 might be causing some issue.

**Reply by Demonese** [reply]
> Are you able to provide some code sample that repros this? I'm not reproing it with

I think I should provide you with the full source file because I can't reproduce the problem anywhere else:

```
add_executable(fontatlas)
set_target_properties(fontatlas PROPERTIES
    MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>"
    C_STANDARD 11
    CXX_STANDARD 20
)
target_compile_options(fontatlas PRIVATE
    "/utf-8"
)
target_include_directories(fontatlas PRIVATE
    ./
)
target_sources(fontatlas PRIVATE
    common.hpp
    common.cpp
    texture.hpp
    texture.cpp
)
target_link_libraries(fontatlas PRIVATE
    windowscodecs.lib
)
```

[src.zip](https://github.com/microsoft/vscode-cpptools/files/6582399/src.zip)

![image](https://user-images.githubusercontent.com/57629614/120433483-dbd72280-c3ad-11eb-82d9-21172c4e53e3.png)
![image](https://user-images.githubusercontent.com/57629614/120433527-eabdd500-c3ad-11eb-9129-59df7c3facdf.png)
![image](https://user-images.githubusercontent.com/57629614/120433673-1e98fa80-c3ae-11eb-959c-31703e8111e3.png)

**Reply by sean-mcmanus** [reply]
The bug repros without our C/C++ extension installed, i.e. the bug is from VS Code itself, specifically the TextMate grammar it uses. I've filed a bug at https://github.com/jeff-hykin/cpp-textmate-grammar/issues/551 .

Also, I reduced the repro to:
```
#define A
struct foo
{
    /*d*/ void func(const char* c)
    {
        #ifdef FOO
            int i;
        #else
            func("" A ")");
        #endif
    }
};
```
